### DISCUSSION TITLE
Question about "Use authenticated user as customer"

### DISCUSSION THREAD

**Discussion by de-raaf-media** [duncanmcclean/simple-commerce#1058] [Feature Requests]
In your commit I see
// When the customer driver is set to users, a user is logged in, and the cart doesn't have a customer,
// we'll set the customer to the logged in user.

What happens (or what should happen) if the user was logged in when adding items to the cart decides to logout to continue as a guest user. I think currently the customer_id will remain attached to the order? Also if the user has another account (edge case?) and logs in with that, the customer_id will remain the same.

First I thought of destroying the cart on logout (still an option) but that doesn't seem how it is generally handled in webshops. Maybe on logout clear the customer_id for the current order? And if you log in with another account update the customer_id to the new user id?

**Comment by duncanmcclean** [comment]
> What happens (or what should happen) if the user was logged in when adding items to the cart decides to logout to continue as a guest user. I think currently the customer_id will remain attached to the order? Also if the user has another account (edge case?) and logs in with that, the customer_id will remain the same.

The logged in user will remain as the customer set on the order. Even if you update the customer information (like their email), it'll update the email of that existing user. 

Now, I think of it, sounds like this feature might have quite a few edge cases I didn't consider. I _may_ end up removing it 🤔

**Reply by de-raaf-media** [reply]
I actually hope you won't remove it ;-)

I am talking about the scenario where a user logs out and then goes to checkout as a guest. Everything will then be updated (which is good) but the customer_id is not cleared. So I think mainly the condition that checks for a user ID should change.

**Reply by duncanmcclean** [reply]
I've just tagged a new release which includes #1068, which'll remove the customer from the order when they logout.
